Consulting 101: Refactor to make the points more concise. (#130)
abtion/guidelines: behaviour_and_protocol/consulting_101.md

  * Your messages are a part of your professional image. Make sure they are thought-through and polite.
* **Ensure a professional setup for remote interactions.**
  * Example: Ensure the background is tidy and well-lit for video meetings.
* **Keep track of time spent on projects accurately and consistently.**
  * Log your hours daily to maintain accurate project timelines and budgeting.

Proactiveness
-----------------------

* **Be proactive in identifying and addressing potential issues or enhancements.**
  * If you find something that could be fixed, could be optimized, or could be updated in other ways, communicate this to the team.
  * Example: "I noticed a strange pattern in the data and wanted to make sure that this is not an error or would otherwise impact the project."
* **Engage in proactive communication while being mindful of the implementation's impact.**
  * Example: Before implementing a change, discuss potential impacts with the client and team.

Solution-Oriented Approach
--------------------------

* **Avoid outright denials when addressing client requests; instead, propose alternative solutions or phased approaches.**
  * Avoid: "That's impossible."

>   * Try: "That's challenging, but let's explore some alternatives that might work."

Reviewer: I'd really like to have the idea of asking the client what they are trying to achieve (when they are suggesting feature) somewhere in the guide, as I think it can be really useful.

I think that maybe this is the best place, but I'm not entirely sure if this suggestion works without changes.

```suggestion
  * Try: "That's challenging. Could you elaborate on the problem you are aiming to solve? Perhaps there are alternatives that might work."
```

  * Avoid: "It will take too much time."
  * Try: "This is a complex feature. If this is critical for the project, we can research and estimate it before we move on."
  * Avoid: "Do you really need that?"
  * Try: "Let us put it in the ice box and assess it when we have finished the business critical features."
* **Suggest a Proof of Concept (POC) for technically challenging or untested solutions to demonstrate if they're feasible.**
  * Example: "We can develop a small POC to test this approach within a week."

Understanding the Client’s Perspective
-------------------------------------------

* **View and assess the platform from the client's perspective to better understand their needs.**
  * You might find new features or improvements that the client has not thought of.

>   * Example: Regularly use the client's platform yourself to identify areas for improvement.

Reviewer: As a developer I'm unsure about what is expected of me here.

Should I regularly just randomly click around the project I'm working on to identify problems?

Author: Good point. 
I have had projects where the team dedicated time to use the system, but it's not a normal thing currently

Author: Do you have any thoughts on how we could clarify?

Author: Good point. I've added your idea with "Test Tuesday"

* **Focus on delivering features that not only meet technical specifications but also help the client's user experience.**

>   * Example: "This feature was designed with your team's workflow in mind, reducing the steps required to achieve the result."

Author: ```suggestion
  * Example: "This feature was updated to speed up your process. It will help your team to achieve results faster."
```

If you are coming from a product company, it can be unclear what consulting is, and what your new role as a consultant requires. Our goal with this document is to provide a clearer overview of consulting, dos and don'ts, etiquette, etc. We believe that training people and bringing them closer to the client ensures that we can give clients better advice and products. As a reminder, please note that the following are guidelines and not strict rules.

## Client communication
- Recognize the client's style of communication. You should not aim to mirror it but it gives you a sense of the tone that they might expect from you.
- We use a digital project management tool (Asana is our default one) as our collaboration platform for prioritizing work with the client. We prefer our clients to be engaged in this collaboration but it is ultimately your responsibility to add and prioritize tasks. Asana is primarily available for the client to provide transparency and as a communication channel. As Asana is public to the client, we expect you to communicate professionally in comments, task descriptions, etc.

> - We want everyone independent of their role to participate actively in client communication both during meetings but also in Slack, Asana, e-mails, or wherever the communication happens. Remember to chat with your team before you reach out to the client. Your colleague might already have asked the same question, and we want to avoid spamming the client with questions. For a more in-depth introduction to how we facilitate team meetings at Abtion, please have a look at [this document](./how_to_facilitate_team_meetings.md).

Reviewer 2: Previously, we had a reference to ./how_to_facilitate_team_meetings.md
Might still be relevant to link to it somewhere.

Reviewer 2: Not merge blocking, the PR looks good.
Just something to consider.

If you are coming from a product company, it can be unclear what consulting is, and what your new role as a consultant requires. Our goal with this document is to provide a clearer overview of consulting. We believe that training people and bringing them closer to the client ensures that we can give clients better advice and products. 

As a reminder, please note that the following are guidelines and not strict rules.

This document is divided into the following sections:
- [Client Communication](#client-communication)
- [Technical Communication](#technical-communication)
- [Etiquette and Professionalism](#etiquette-and-professionalism)
- [Proactiveness](#proactiveness)
- [Solution-Oriented Approach](#solution-oriented-approach)
- [Understanding the Client’s Perspective](#understanding-the-clients-perspective)
- [Client-Specific Insights](#client-specific-insights)

Client Communication
--------------------

* **Recognize and understand the client's preferred communication style.**
  * If a client prefers to understand the context for your solutions in detail, provide comprehensive explanations.
* **Keep communication transparent and public.**

>   * When communicating about a specific task, be sure to keep the communication in a shared platform like Asana or Slack. Ideally on the task itself.

Reviewer 3: Swapping Asana and Slack would make it more congruent with the following sentence.

Technical Communication
----------------------------------

* **Tailor communication to the client's technical understanding.**
  * Ask the client how technical they would like the explanation to be.
  * If the client is non-technical, keep the explanations simple and eg. use metaphors.
* **Provide realistic estimates and pose potential challenges and solutions.**
  * Example: "We have estimated the SSO login as an XL-task. The documentation for the SSO is new to us, so we would have to spend some time researching. Alternatively, an email-based login would be an L-task."
* **Explain system issues, highlighting the reasons for unexpected outcomes and the steps taken to fix these.**
  * Example: "The issue occurred because the user could input text, and we were expecting numbers. We've updated the input and added additional tests."

Etiquette and Professionalism
-----------------------------

* **Punctuality for meetings.**
  * Some meetings require more preparation than others. Make sure to prepare for the meeting.
  * Example: Arrive 5-10 minutes early to setup and review meeting materials.
* **Stay professional in both physical and digital interactions.**

>   * You do not need formal attire for client meetings, but you should be presentable.

Author: ```suggestion
  * You do not need formal attire for client meetings, but you should be presentable. This also applies to your background in online meetings.
```

>   * You do not need formal attire for client meetings, but you should be presentable. This also applies to your background in online meetings.

Author: ```suggestion
  * You do not need formal attire for client meetings, but you should be presentable.
```

  * Your messages are a part of your professional image. Make sure they are thought-through and polite.
* **Ensure a professional setup for remote interactions.**
  * Example: Ensure the background is tidy and well-lit for video meetings.
* **Keep track of time spent on projects accurately and consistently.**

>   * Example: Log your hours daily to maintain accurate project timelines and budgeting.

Author: ```suggestion
  *Log your hours daily to maintain accurate project timelines and budgeting.
```

* **Stay professional in both physical and digital interactions.**
  * You do not need formal attire for client meetings, but you should be presentable.
  * Your messages are a part of your professional image. Make sure they are thought-through and polite.
* **Ensure a professional setup for remote interactions.**
  * Example: Ensure the background is tidy and well-lit for video meetings.
* **Keep track of time spent on projects accurately and consistently.**
  * Log your hours daily to maintain accurate project timelines and budgeting.

Proactiveness
-----------------------

* **Be proactive in identifying and addressing potential issues or enhancements.**
  * If you find something that could be fixed, could be optimized, or could be updated in other ways, communicate this to the team.
  * Example: "I noticed a strange pattern in the data and wanted to make sure that this is not an error or would otherwise impact the project."
* **Engage in proactive communication while being mindful of the implementation's impact.**
  * Example: Before implementing a change, discuss potential impacts with the client and team.

Solution-Oriented Approach
--------------------------

> * **Avoid outright denials when addressing client requests; instead, propose alternative solutions or phased approaches.**
>   * Avoid: "That's impossible."
>   * Try: "That's challenging, but let's explore some alternatives that might work."

Author: ```suggestion
* **Avoid outright denials when addressing client requests; instead, propose alternative solutions or phased approaches.**
  * Avoid: "That's impossible."
  * Try: "That's challenging, but let's explore some alternatives that might work."
  * Avoid: "It will take too much time."
  * Try: "This is a complex feature. If this is critical for the project, we can research and estimate it before we move on."
```

  * Example: Ensure the background is tidy and well-lit for video meetings.
* **Keep track of time spent on projects accurately and consistently.**
  * Log your hours daily to maintain accurate project timelines and budgeting.

Proactiveness
-----------------------

* **Be proactive in identifying and addressing potential issues or enhancements.**
  * If you find something that could be fixed, could be optimized, or could be updated in other ways, communicate this to the team.
  * Example: "I noticed a strange pattern in the data and wanted to make sure that this is not an error or would otherwise impact the project."
* **Engage in proactive communication while being mindful of the implementation's impact.**
  * Example: Before implementing a change, discuss potential impacts with the client and team.

Solution-Oriented Approach
--------------------------

* **Avoid outright denials when addressing client requests; instead, propose alternative solutions or phased approaches.**
  * Avoid: "That's impossible."
  * Try: "That's challenging, but let's explore some alternatives that might work."
  * Avoid: "It will take too much time."

>   * Try: "This is a complex feature. If this is critical for the project, we can research and estimate it before we move on."

Author: ```suggestion
  * Try: "This is a complex feature. If this is critical for the project, we can research and estimate it before we move on."
  * Avoid: "Do you really need that?"
  * Try: "Let us put it in the ice box and assess it when we have finished the business critical features."
```
